File: AmbientZero.Web/AmbientZero.Web.Client/Models/SensorInformation.cs
namespace AmbientZero.Web.Client.Models
{
    public class SensorInformation
    {
        public int Id { get; set; }
        public string DeviceId { get; set; }
        public string? DeviceVirtualName { get; set; }
        public int? ADCReading { get; set; }
        public bool? IsSignaling { get; set; }
        public DateTime LastReadingTime { get; set; }
        public float? LastReadingValue { get; set; }
        public string TemperatureTendency { get; set; }
        public SensorSettings SensorSetting { get; set; }
        public string DisplayName => string.IsNullOrEmpty(DeviceVirtualName) ? $"{DeviceId}*" : DeviceVirtualName;
        public double CleanTemperature => LastReadingValue.HasValue ? Math.Round(LastReadingValue!.Value, 1) : 0;
    }
}

File: src/app/interfaces/sensor-information.ts
import { SensorSetting } from "./sensor-setting";

export interface SensorInformation {
    id: number,
    deviceId : string,
    deviceVirtualName: string|null,
    adcReading: number,
    isSignaling: boolean,
    lastReadingTime: string,
    lastReadingValue: number,
    temperatureTendency: string,
    sensorSetting: SensorSetting
}

